proxy/src/cancellation.rs

```diff
@@ -29,14 +32,23 @@ pub struct CancellationHandler<P> {
     /// This field used for the monitoring purposes.
     /// Represents the source of the cancellation request.
     from: CancellationSource,
+    // rate limiter of cancellation requests
+    limiter: Arc<Mutex<GlobalRateLimiter>>,
```

Comment:
This can be a std Mutex. (Ideally, the rate limiter should not need a mutex.)

Reply:
We also need to think about fairness. A user DoSing might bury a valid cancel request by someone else with a truly fair mutex, which I think tokio's is.

Reply:
good point, thanks!